dbi/src/main/java/com/github/freeacs/dbi/ACS.java

```diff
@@ -262,15 +262,11 @@ private void readUnittypeParameterValues(Unittypes unittypes) throws SQLExceptio
     Statement s = null;
     ResultSet rs = null;
     String sql;
-    boolean wasAutoCommit = false;
-    Connection connection = null;
-    try {
+    try(Connection connection = getDataSource().getConnection()) {
       sql = "SELECT utp.unit_type_id, utpv.unit_type_param_id, value, priority, type ";
       sql += "FROM unit_type_param_value utpv, unit_type_param utp ";
       sql += "WHERE utpv.unit_type_param_id = utp.unit_type_param_id ";
       sql += "ORDER BY utp.unit_type_id ASC, utpv.unit_type_param_id, utpv.priority ASC";
-      connection = getDataSource().getConnection();
-      wasAutoCommit = connection.getAutoCommit();
       connection.setAutoCommit(false);
       s = connection.createStatement();
       s.setQueryTimeout(60);
```

Comment:
can you apply this for me

Reply:
There is basically littered around this kind of code  this .setQueryTimeout(<number>) and .setAutoCommit(true|false)

what is your advice on how to solve this. Do we really need fine grained control on query timeouts? do we actually need to control auto commit ? If we had used transactions, the transaction would take care of it right?

Reply:
Is it important when using hikari connection pooling, to reset auto commit to the previous value ?

Reply:
ResultSets and query timeouts doesn't need to be reset right? because ResultSets are transient objects ?

Reply:
> @jarlah, to reset the auto-commit mode to its previous value when using a try-with-resources block, you can use a finally block outside the try-with-resources statement. Here's how you can structure your code to achieve this:
> 
> ```java
> boolean originalAutoCommit = true;
> Connection connection = null;
> try {
>     connection = getDataSource().getConnection();
>     originalAutoCommit = connection.getAutoCommit();
>     connection.setAutoCommit(false);
>     try (Statement statement = connection.createStatement();
>          ResultSet resultSet = statement.executeQuery("your query here")) {
>         // Use the connection, statement, and result set
>     } // try-with-resources will automatically close statement and resultSet
> } catch (SQLException e) {
>     // Handle exceptions
> } finally {
>     if (connection != null) {
>         try {
>             connection.setAutoCommit(originalAutoCommit); // Reset auto-commit
>             connection.close(); // Explicitly close connection
>         } catch (SQLException e) {
>             // Handle potential exception during close
>         }
>     }
> }
> ```
> 
> This approach ensures that you manually reset the auto-commit mode to its original value before closing the connection, even when using try-with-resources for managing other resources like `Statement` and `ResultSet`.

what about the try with resources ? is it not possible to combine overriding auto commit AND use try with resources?

Reply:
I did this 

```java
public class AutoCommitResettingConnectionWrapper implements AutoCloseable {
    private final Connection connection;
    private final boolean originalAutoCommit;

    public AutoCommitResettingConnectionWrapper(Connection connection, boolean autoCommit) throws SQLException {
        this.connection = connection;
        this.originalAutoCommit = connection.getAutoCommit();
        connection.setAutoCommit(autoCommit);
    }

    public Connection getConnection() {
        return connection;
    }

    @Override
    public void close() throws SQLException {
        try {
            connection.setAutoCommit(originalAutoCommit);
        } finally {
            connection.close();
        }
    }
}
```

because im so lazy could you make a similar version for Statement, that takes in the connection and query timeout ? but this time, its not necessary to revert the original query timeout of course

Reply:
@coderabbitai I need help with the readHeartbeats method in ACS class. It's using a DynamicStatement, which is a custom class for building up sql queries. The problem is that I might need to make a different wrapper for this class ?  Where its constructor takes in the connection, passes the provided sql string into addSqlAndArguments and then executes the method makePreparedStatement, for ex when calling getPreparedStatement in the wrapper?